ClientPatcher/SettingsManager.cs

```diff
@@ -52,52 +59,53 @@ public List<PatcherSettings> GetNewSettings()
             catch (Exception e)
             {
                 throw new InvalidOperationException("Unable to download settings from server." + e);
-            }         
+            }
         }
+
         /// <summary>
         /// Merges a list of PatcherSettings objects with the currently loaded settings
         /// </summary>
         /// <param name="webSettingsList">List of PatcherSetting objects to merge</param>
-        public void MergeWebSettings(List<PatcherSettings> webSettingsList )
+        public void MergeWebSettings(List<PatcherSettings> webSettingsList)
         {
             if (Servers == null)
             {
                 Servers = webSettingsList;
                 return;
             }
-            foreach (PatcherSettings webProfile in webSettingsList) //Loop through loaded settings from settings.txt
+
+            // Loop through loaded settings from settings.txt.
+            foreach (PatcherSettings webProfile in webSettingsList)
             {
-                //find the matching local profile by Guid
+                // Find the matching local profile by Guid.
                 PatcherSettings localProfile = Servers.FirstOrDefault(i => i.Guid == webProfile.Guid);
-                //if a local match, update, else, add a new local profile
+
+                // If a local match, update, else, add a new local profile.
                 if (localProfile != null)
                 {
                     if (!webProfile.DeleteProfile)
                     {
-                        //dont update all fields on purpose. user retains control of paths, which client to use.
-                        localProfile.PatchBaseUrl = webProfile.PatchBaseUrl;
-                        localProfile.PatchInfoUrl = webProfile.PatchInfoUrl;
-                        localProfile.ServerName = webProfile.ServerName;
-                        localProfile.FullInstallUrl = webProfile.FullInstallUrl;
-                        localProfile.AccountCreationUrl = webProfile.AccountCreationUrl;
-                        localProfile.Enabled = webProfile.Enabled;
+                        // Don't update all fields on purpose. user retains control of paths, which client to use.
+                        UpdateProfile(localProfile, webProfile);
                     }
                     else
                     {
-                        // remove from the list of servers (will also not be in saved list)
+                        // Remove from the list of servers (will also not be in saved list).
                         Servers.Remove(localProfile);
                     }
                 }
                 else
                 {
-                    // don't add a server that is supposed to be deleted
+                    // Don't add a server that is supposed to be deleted.
                     if (!webProfile.DeleteProfile)
                         Servers.Add(webProfile);
                 }
```

Comment:
after the end of this if() statement, we can run another search in local profiles, but instead of matching on Guid, match on the patchinfourl: 
localProfile = Servers.FirstOrDefault(i => i.PatchInfoUrl == webProfile.PatchInfoUrl);
This is how we can locate a profile based on something other than a Guid for updating old profiles